Sources/Fuzzilli/Mutators/OperationMutator.swift

```diff
@@ -119,22 +135,20 @@ public class OperationMutator: BaseInstructionMutator {
             newOp = ReassignWithBinop(chooseUniform(from: allBinaryOperators))
         case let op as DestructArray:
             var newIndices = Set(op.indices)
-            if newIndices.count > 0 {
-                newIndices.remove(newIndices.randomElement()!)
-                while newIndices.count != op.indices.count {
-                    newIndices.insert(Int.random(in: 0..<10))
-                }
-            }
+            replaceRandomElement(in: &newIndices, generatingRandomValuesWith: { return Int.random(in: 0..<10) })
             newOp = DestructArray(indices: newIndices.sorted(), hasRestElement: !op.hasRestElement)
         case let op as DestructArrayAndReassign:
             var newIndices = Set(op.indices)
-            if newIndices.count > 0 {
-                newIndices.remove(newIndices.randomElement()!)
-                while newIndices.count != op.indices.count {
-                    newIndices.insert(Int.random(in: 0..<10))
-                }
-            }
+            replaceRandomElement(in: &newIndices, generatingRandomValuesWith: { return Int.random(in: 0..<10) })
             newOp = DestructArrayAndReassign(indices: newIndices.sorted(), hasRestElement: !op.hasRestElement)
+        case let op as DestructObject:
+            var newProperties = Set(op.properties)
+            replaceRandomElement(in: &newProperties, generatingRandomValuesWith: { return b.type(of: instr.input(0)).properties.randomElement() ?? b.genPropertyNameForRead() })
```

Comment:
Let's just use `b.genPropertyNameForRead()`. The issue is that if you have an object for which AI knows only one property, then this mutation will just pick that one property 5 times, then give up. So then I think it's better to just choose completely random property names, which we also do elsewhere in this mutator (instead of relying on type information).